src/elements/TextInput/TextInput.stories.tsx

```diff
@@ -1,28 +1,97 @@
 /* (c) Copyright Frontify Ltd., all rights reserved. */
 
+import IconIcons from "@elements/Icon/Generated/IconIcons";
+import { IconSize } from "@elements/Icon/Icon";
 import { Meta, Story } from "@storybook/react";
-import TextInputComponent, { TextInputProps, TextInputType } from "./TextInput";
+import TextInput, { TextInputProps, TextInputType, Validation } from "./TextInput";
 
 export default {
     title: "Elements/Text Input",
-    component: TextInputComponent,
+    component: TextInput,
     args: {
         clearable: false,
         disabled: false,
         required: false,
+        dotted: false,
         type: TextInputType.Text,
+        validation: Validation.Default,
     },
     argTypes: {
-        type: {
-            options: Object.values(TextInputType),
+        validation: {
+            options: Object.values(Validation),
             type: "select",
         },
-        decorator: { type: "string" },
         placeholder: { type: "string" },
-        onInput: { action: "onInput" },
-        onBlur: { action: "onBlur" },
-        onClear: { action: "onClear" },
+        defaultValue: { type: "string" },
+        obfuscated: { table: { disable: true } },
+        type: { table: { disable: true } },
+        decorator: { table: { disable: true } },
+        onInput: { table: { disable: true }, action: "onInput" },
+        onBlur: { table: { disable: true }, action: "onBlur" },
+        onClear: { table: { disable: true }, action: "onClear" },
     },
 } as Meta<TextInputProps>;
 
-export const TextInput: Story<TextInputProps> = (args: TextInputProps) => <TextInputComponent {...args} />;
+const TextInputTemplate: Story<TextInputProps> = (args) => <TextInput {...args} />;
+
+export const Text = TextInputTemplate.bind({});
+
+export const Prefilled = TextInputTemplate.bind({});
+
+Prefilled.args = {
+    defaultValue: "Value text",
```

Comment:
Somehow when a `defaultValue` is set, it doesn't get updated if it's changed via the storybook control.

Reply:
since the underlying `input` does not re-render if it is initially set with a `defaultValue` this does not work as expected i changed it to only allow to change the `defaultValue` if it is not set initially see b3ee7fc